test_decoding.py

```diff
@@ -91,14 +107,148 @@
         # save data in the data dictionary, for later access
         data[f'{subj}-{night}'] = times, data_x, data_y
 
+
         for which in ['before', 'after']: # also load the data of the test session
+        
             times, data_x, data_y  = load_test(folder_night, which=which, sfreq=sfreq,
                                                tmin=tmin, tmax=tmax, event_id=trigger.STIMULUS)
 
             data_test[f'{subj}-{night}-{which}'] = times, data_x, data_y
+            
+        
     # except:
         # print(f'ERROR: {folder}')
 
+#%% Mondo: Input Data Preparation
+
+# List of data names from all session
+data_name = list(data_test.keys())
+
+# Take example data from subj 1 (PN01-high-before)
+X_times, X_PN01_hb, y = data_test[data_name[18]]
+
+# Label dataset in dataframe format
+y_df = pd.DataFrame(y)
+
+# Some ways to get the column name (label name)
+# y_col_df = sorted(y_df) #returns the label name alphabetically
+# y_col_df = y_df.columns.values.tolist()
+# y_col = list(y.keys()) #returns the label name directly from dict
+
+# Separating new/old image class
+# y_img_truth = data_label['seen_before_truth']
+# y_img_resp = data_label['seen_before_response']
+y_img_truth = y_df.seen_before_truth.replace({True: 1, False: 0})
+y_img_resp = y_df.seen_before_response.replace({True: 1, False: 0})
+
+# Need to balance classes
+
+#%% Assemble Classifier
+
+# Feature Extraction
+features_example = {'mean', 'ptp_amp', 'std'} # func_params
+X_PN01_hb = extract_features(X_PN01_hb, sfreq, features_example) #returns (n_epochs, n_features)
+
+logreg = LogisticRegression(penalty='l1', C=1/1, solver='liblinear')
+# lda = LinearDiscriminantAnalysis()
+# csp = CSP(n_components=4, reg=None, log=True, norm_trace=False)
+
+# # Dimensionality Reduction (PCA)
+# pca = PCA(n_components=0.95)
+# X_PN01_hb = pca.fit_transform(X_PN01_hb)
+
+# Make Pipeline
+pipe_1 = Pipeline([('scaler', StandardScaler()), ('lr', logreg)])
+# pipe_2 = Pipeline([('CSP', csp), ('LDA', lda)])

```

Comment:
Not sure which scaler to use, what do you thin?.

Reply:
I think the StandardScaler is good for now